### PR TITLE
feat: add sha256 to OTP builds.txt

### PR DESCRIPTION
**Pull request by gigatim**
This PR adds a sha256 to the builds.txt files for OTP.

## Background 

PR #34 added sha256 to the elixir builds.txt  [This comment](https://github.com/hexpm/bob/pull/34#issuecomment-564041915) called out that a separate PR should be created to do the same to the OTP builds.txt file.  It appears this was never completed.

Also relates to #16

## Test Disclaimer

I was not able to run it locally. Therefore the PR is provided without any testing at all.

### CODE REVIEW

**Comment by gigatim** [comment]
@ericmj Can I get a review on this PR?  I do not have the permissions to request a reviewer in the GitHub UI.

File: priv/scripts/otp/otp.sh
Comment (wojtekmach) [inline priv/scripts/otp/otp.sh:49]: since we didn't have this field yet, I'd use this opportunity to make it super explicit what it is:

```suggestion
echo -e "${ref_name} ${ref} $(date -u '+%Y-%m-%dT%H:%M:%SZ') sha256:${build_sha256}\n$(cat builds.txt)" > builds.txt
```

That being said, elixir builds.txt does NOT have this `sha256:` prefix so perhaps it's better to keep it consistent.

**Review by wojtekmach — APPROVED** [review]

File: priv/scripts/otp/otp.sh
Comment (gigatim) [reply priv/scripts/otp/otp.sh:49]: I agree, the tag would be nice.  However, I think consistency would be more important at this point.

**Comment by wojtekmach** [comment]
Before merging this, I think we should have a script ready that updates all existing builds.txt with the checksums. WDYT @ericmj?

These are all the files we need to update:

```
$ for arch in amd64 arm64; do for os in ubuntu-{20.04,22.04,24.04}; do echo https://builds.hex.pm/builds/otp/$arch/$os/builds.txt; done ; done
https://builds.hex.pm/builds/otp/amd64/ubuntu-20.04/builds.txt
https://builds.hex.pm/builds/otp/amd64/ubuntu-22.04/builds.txt
https://builds.hex.pm/builds/otp/amd64/ubuntu-24.04/builds.txt
https://builds.hex.pm/builds/otp/arm64/ubuntu-20.04/builds.txt
https://builds.hex.pm/builds/otp/arm64/ubuntu-22.04/builds.txt
https://builds.hex.pm/builds/otp/arm64/ubuntu-24.04/builds.txt
```

**Comment by gigatim** [comment]
> Before merging this, I think we should have a script ready that updates all existing builds.txt with the checksums. WDYT

I disagree. This is adding additional meta data, which does not necessarily need to be retroactively added.  I'm not opposed to having older versions updated, but I don't know that this PR should be blocked on that work.

For instance, the https://builds.hex.pm/builds/elixir/builds.txt only has this additional field on some (newer) lines.

**Comment by wojtekmach** [comment]
Thank you!

**Pull request merged**
